frontend/src/components/Forms/LoginPage/LoginPage.tsx

```diff
@@ -0,0 +1,125 @@
+import { useRef, useState } from "react";
+import classes from "../Forms.module.scss";
+
+function LoginPage({ onClick }: { onClick: () => void }) {
+  const [formData, setFormData] = useState({
+    username: "",
+    password: "",
+  });
+  const [usernameFound, setUsernameFound] = useState(true);
+  const [passwordMatch, setPasswordMatch] = useState(true);
+  const [usernameEmpty, setUsernameEmpty] = useState(false);
+  const [passwordEmpty, setPasswordEmpty] = useState(false);
+  const [isProcessing, setIsProcessing] = useState(false);
+  const usernameField = useRef<HTMLInputElement>(null);
+  const passwordField = useRef<HTMLInputElement>(null);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+    if (!usernameFound) setUsernameFound(true);
+    if (!passwordMatch) setPasswordMatch(true);
+    if (usernameEmpty) setUsernameEmpty(false);
+    if (passwordEmpty) setPasswordEmpty(false);
+    usernameField.current?.classList.remove(classes.inputError);
+    passwordField.current?.classList.remove(classes.inputError);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (isProcessing) return;
+    setIsProcessing(true);
+
+    if (formData.username === "") {
+      usernameField.current?.classList.add(classes.inputError);
+      return setUsernameEmpty(formData.username === "");
+    }
+    if (formData.password === "") {
+      passwordField.current?.classList.add(classes.inputError);
+      return setPasswordEmpty(formData.password === "");
+    }
+
+    try {
+      const resp = await fetch("http://localhost:3000/login", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      // need backend for auth/login to display errors for username not found / password invalid.
+      if (resp.ok) {
+        console.log("success");
+      } else {
+        console.log("failure");
+      }
+    } catch (err) {
+      console.log("Error: ", err);
+    } finally {
+      setIsProcessing(false);
+    }
+  };
+
+  return (
+    <div className={classes.container}>
+      <div className={classes.sheet}>
+        <h1 className={classes.title}>Welcome!</h1>
+        <form className={classes.form} onSubmit={handleSubmit}>
+          <span className={classes.label}>Username</span>
+          <input
+            ref={usernameField}
```

Comment:
this is a subjective matter, but might i suggest adding the error classes with state instead of refs? the benefit of this would be that you can specify the error-styling behaviour declaratively in the jsx rather than elsewhere

Reply:
I'm confused on how ternary operators would work for single if statements 
```js
<condition> ? true : false
<state> ? className={class.error} : ?
```

Reply:
this sort of thing is pretty popular:
`<input className={ state ? class.error : '' } />`